### DISCUSSION TITLE
请问iOS如何自己打包出xcframework使用

### DISCUSSION THREAD

**Discussion by swlfigo** [Tencent/libpag#372] [Q&A] [answered]
比如改了源码，想重新打包一个xcframework(非.a,如podfile下载的xcframework)，应该如何操作

**Comment by kevingpqi123** [comment, marked as answer]
基于源码目录下的 ios 的 demo 工程就可以，可以看下这篇文档：
https://www.jianshu.com/p/437b1935c6c2

**Comment by swlfigo** [comment]
你好想问一下，基于ios demo的工程，执行完 pod install 之后他会在pod下生成xcframework了，看了下是podspec里面的
system("node build_vendor #{vendorNames} -o #{PAG_ROOT}/ios/Pods/pag-vendor -p ios --xcframework")
这个开始cmake编译，想问下能否生成不同架构的.a 和符号表?

**Reply by kevingpqi123** [reply]
可以不用管这里，如果需要编译 .a, 你还是基于 XCode 的工程，配置生成静态库 .a 文件，然后还是通过 XCodeBuild, 获取库文件和符号表

**Comment by swlfigo** [comment]
感谢答复。我尝试使用 node build_vendor -o /path -p ios --xcframework , 生成xcframework的话，是没有header的，如果要像pod install完生成的xcframework带header的话，是不是需要 build_vendor 时候不生成xcframework，选择生成不同平台的.a, 然后新建空xcode 工程，手动link .a,添加头文件再自己打包生成framework,最后link再一起成为xcframework呢？

**Reply by kevingpqi123** [reply]
静态库版本的编译会麻烦一些，如果没有特殊需求，建议直接使用动态库，如果需要使用，在我们的版本发布页面提供的也有静态库的版本
关于静态库的编译，通过 xcodebuild 编译 的时候加上这些配置就可以 'MACH_O_TYPE=staticlib GCC_DEBUGGING_SYMBOLS=used GCC_GENERATE_DEBUGGING_SYMBOLS=NO'
另外静态库的编译和动态库不太一样：静态库需要和自身依赖的静态库做合并（XCFrameworkIntermediates/libpag/libpag-vendor.a），并且需要先拆分架构然后做合并，单架构完成合并后，然后再把四个架构合并在一起，这些内容都是通用的编译方法，你可以 google 下相关内容
